airflow/utils/helpers.py

```diff
@@ -258,8 +257,7 @@ def build_airflow_url_with_query(query: Dict[str, Any]) -> str:
     import flask
 
     view = conf.get('webserver', 'dag_default_view').lower()
-    url = flask.url_for(f"Airflow.{view}")
-    return f"{url}?{parse.urlencode(query)}"
+    return flask.url_for(f"Airflow.{view}", **query)
```

Comment:
Turns out URL for handles url prams and everything else gets set as query params, so the fix was easy :)